internal/executor/executor.go

```diff
@@ -604,6 +604,12 @@ func (executor *Executor) CloneRepository(
 			fetchOptions.Depth = clone_depth + 1
 		}
 		err = repo.FetchContext(ctx, fetchOptions)
+		if err != nil && strings.Contains(err.Error(), "couldn't find remote ref") {
+			logUploader.Write([]byte("\nFailed to fetch merge ref! PR might not be mergable. Falling back to head ref..."))
+			headRefSpec := fmt.Sprintf("+refs/pull/%s/head:refs/remotes/origin/pull/%[1]s", pr_number)
+			fetchOptions.RefSpecs = []config.RefSpec{config.RefSpec(headRefSpec)}
+			err = repo.Fetch(fetchOptions)
```

Comment:
Shouldn't we use `FetchContext()` here too?

Reply:
I copied the logic from below retry logic. 🤔 Will check if we need to change it in both places.

Reply:
Fixed in [61b473b](https://github.com/cirruslabs/cirrus-ci-agent/pull/250/commits/61b473bd6581783a2c129fba7665c281834ff2ee)